### PR TITLE
BJ-Share: use cookie login method. resolves #12756

### LINKED ISSUE #12756: Indexer does not accept login without Captcha
### Have you checked our Troubleshooting page for your issue?

- [X] I have checked the Troubleshooting page

### Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### Have you read our Contributing Guidelines?

- [X] I have read the Contributing Guidelines

### Environment

```markdown
Using HTTP Client: HttpWebClient2
Using FlareSolverr: No
Using proxy: Disabled
App config/log directory: /home/raphaelmaria/.config/Jackett
ThreadPool MaxThreads: 32767 workerThreads, 1000 completionPortThreads
Running in Docker: No
File /etc/issue: Ubuntu 20.04.3 LTS \n \l
Jackett variant: CoreLinuxAmdx64
OS version: Unix 5.11.0.43 (64bit OS) (64bit process)
Environment version: 6.0.1 (/opt/Jackett/)
Starting Jackett v0.20.216
```

### Description

On today's date, some of my indexes started asking for revalidation, asking for a captcha, but in the window that opens next, it doesn't edit at all and doesn't answer the open fields, is there any paleative for that?

### Logged Error Messages

Jackett.Common.ExceptionWithConfigData: <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="pt_BR" lang="pt_BR">
<head>
	<title>Login :: BJ-Share :: Semeando Amizades</title>
	<meta name="robots" content="noindex,nofollow">
	<meta http-equiv="X-UA-Compatible" content="chrome=1; IE=edge" />
	<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
	<link rel="shortcut icon" href="favicon.ico" />
	<link href="static/styles/public/style.css?v=1640813253" rel="stylesheet" type="text/css" />
	<script src="static/functions/jquery.js"></script>
	<script src="static/functions/script_start.js?v=1619390454"></script>
	<script src="static/functions/ajax.class.js?v=1599349412"></script>
	<script src="static/functions/global.js?v=1630449800"></script>
</head>
<body>
    <div align="center">
            <a href="login.php"><div id="logo"></div></a>
<span id="no-cookies" class="hidden warning">Seus cookies estão desativados.<br/><br/></span>
<noscript><span class="warning">BJ-Share precisa do JavaScript para funcionar. Ative o JavaScript no seu navegador.</span><br/><br/>
</noscript>
    <form class="auth_form" name="xalogin" id="xaloginform" method="post" action="login.php" autocomplete="off">
        <div id="msgsPublic">
                            <span class="warning">Você precisa validar o Recaptcha, tente novamente.<br/><br/></span>
        </div>
        <div id="publicFormDiv">
            <input type="text" maxlength="25" name="username" id="username" autofocus="autofocus" placeholder="Usuário" />
            <input type="password" name="password" id="password" placeholder="Senha" />

                        <br />
            <input type="checkbox" id="keeplogged" name="keeplogged" value="1" checked="checked" />
            <label for="keeplogged">Lembrar-me</label><br/><br/>
    <script src="https://www.google.com/recaptcha/api.js" async defer></script>
    <div class="g-recaptcha" data-sitekey="6LcDdqsZAAAAAASBVUYO3WoJ2pX2cuzqJwemkJoC" data-theme="dark"></div>
    <br />
            <input type="submit" value="Login" class="submit"/>
        </div>
    </form>
                <div id="bottomTexts">
            <br><a href="login.php?act=recover" class="tooltip" title="Recuperar Senha">Recuperar senha</a>        </div>
<script>
    cookie.set('cookie_test', 1, 1);
    if (cookie.get('cookie_test') != null) {
        cookie.del('cookie_test');
    } else {
        $('#no-cookies').gshow();
    }
</script>
<div>
    <span><a href="https://bj-share.info/">BJ-Share &copy; 2016 - 2021</a></span>
</div>
</div>
</body>
</html>

   at Jackett.Common.Indexers.BJShare.<>c__DisplayClass17_0.<ApplyConfiguration>b__0() in /home/vsts/work/1/s/src/Jackett.Common/Indexers/BJShare.cs:line 137
   at Jackett.Common.Indexers.BaseIndexer.ConfigureIfOK(String cookies, Boolean isLoggedin, Func`1 onError) in /home/vsts/work/1/s/src/Jackett.Common/Indexers/BaseIndexer.cs:line 258
   at Jackett.Common.Indexers.BJShare.ApplyConfiguration(JToken configJson) in /home/vsts/work/1/s/src/Jackett.Common/Indexers/BJShare.cs:line 133
   at Jackett.Server.Controllers.IndexerApiController.UpdateConfig(ConfigItem[] config) in /home/vsts/work/1/s/src/Jackett.Server/Controllers/IndexerApiController.cs:line 97
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|20_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jackett.Server.Middleware.CustomExceptionHandler.Invoke(HttpContext httpContext) in /home/vsts/work/1/s/src/Jackett.Server/Middleware/CustomExceptionHandler.cs:line 26

### Screenshots

Screen Shot 2021-12-31 at 00 25 05

### PR DESCRIPTION
**Pull request by uilton-oliveira**
The tracker is now using reCaptcha on every login, so switching to cookie based login will fix it.
Also added "E" prefix to animes episodes with absolute numbering, fix the One Piece episodes > 1000 detection on Sonarr.

Fixes #12756

### CODE REVIEW

**Comment by uilton-oliveira** [comment]
Looking at failed tests, it looks like that it is not related to this PR, right?

**Comment by ilike2burnthing** [comment]
Yea, that's not your PR, one of our tests was expecting to see `2019` as a response:
https://github.com/Jackett/Jackett/blob/30b30e886124ba88e4d03351a87430d7f88a54db/src/Jackett.Test/Common/Indexers/CardigannIndexerHtmlTests.cs#L58
but it's based on a `timeago` field:
https://github.com/Jackett/Jackett/blob/2bf352603389d49386c67b82e5edeca3d1769010/src/Jackett.Test/Resources/html-response1.html#L89
and seemingly new years eve was early enough to cause an issue.

**Comment by ilike2burnthing** [comment]
Thanks. That will be published in ~11hrs from now, depending on how many drinks we have at the Jackett office new years party 😉

**Comment by uilton-oliveira** [comment]
Thanks, happy new year :)
